src/commands.rs

```diff
@@ -4179,11 +4179,12 @@ fn cmd_git_push(
             &args.remote
         ));
         for (change_str, commit) in std::iter::zip(args.change.iter(), commits) {
-            let branch_name = format!(
-                "{}{}",
-                command.settings().push_branch_prefix(),
-                commit.change_id().hex()
-            );
+            // Use the short hash when it's not ambiguous (which it shouldn't be most of the time).
+            let mut change_id = short_change_hash(commit.change_id());
+            if workspace_command.resolve_single_rev(&change_id).is_err() {
+                change_id = commit.change_id().hex();
+            }
+            let branch_name = format!("{}{}", command.settings().push_branch_prefix(), change_id);
```

Comment:
I just realized that this will mean that if an existing user has a branch with the longer name and they then upgrade and run `jj git push --change` again on the same change, they'll get a new branch, right? Should we check if the long branch name already exists and use that name if it does?

Reply:
Ah, sorry! Good catch, I moved this logic to the branch creation arm to fix this (and added a test case). PTAL!